plugins/hypervisors/vmware/test/com/cloud/hypervisor/vmware/resource/VmwareResourceTest.java

```diff
@@ -117,4 +154,79 @@ public void testStartVm3dgpuEnabled() throws Exception{
         verify(vmMo3dgpu).configureVm(any(VirtualMachineConfigSpec.class));
     }
 
-}
+    // ---------------------------------------------------------------------------------------------------
+
+    @Test
+    public void testgetNfsVersionFromNfsTONull(){
+        assertFalse(_resource.getStorageNfsVersionFromNfsTO(null));
+    }
+
+    @Test
+    public void testgetNfsVersionFromNfsTONfsVersionNull(){
+        when(srcDataNfsTO.getNfsVersion()).thenReturn(NFS_VERSION_NOT_PRESENT);
+        assertFalse(_resource.getStorageNfsVersionFromNfsTO(srcDataNfsTO));
+    }
+
+    @Test
+    public void testgetNfsVersionFromNfsTONfsVersion(){
+        assertTrue(_resource.getStorageNfsVersionFromNfsTO(srcDataNfsTO));
+    }
+
+    // ---------------------------------------------------------------------------------------------------
+
+    @Test
+    public void testSetCurrentNfsVersionInProcessorAndHandler(){
+        _resource.setCurrentNfsVersionInProcessorAndHandler();
+        verify(storageHandler).reconfigureNfsVersion(any(Integer.class));
+    }
+
+    // ---------------------------------------------------------------------------------------------------
+
+    @Test
+    public void testExamineStorageSubSystemCommandNfsVersionNotPresent(){
+        when(srcDataNfsTO.getNfsVersion()).thenReturn(NFS_VERSION_NOT_PRESENT);
+        _resource.examineStorageSubSystemCommandNfsVersion(storageCmd);
+        verify(_resource, never()).setCurrentNfsVersionInProcessorAndHandler();
+    }
+
+    @Test
+    public void testExamineStorageSubSystemCommandNfsVersion(){
+        _resource.examineStorageSubSystemCommandNfsVersion(storageCmd);
+        verify(_resource).setCurrentNfsVersionInProcessorAndHandler();
+    }
+
+    // ---------------------------------------------------------------------------------------------------
+
+    @Test
+    public void testStorageSetNfsVersion(){
+        _resource.checkStorageProcessorAndHandlerNfsVersionAttribute(storageCmd);
+        assertEquals(NFS_VERSION, _resource._storageNfsVersion);
+
+        //Call again to verify that version won't be set again once it is set (examineStorageSubSystemCommandNfsVersion - only one time)
+        _resource.checkStorageProcessorAndHandlerNfsVersionAttribute(storageCmd);
+        verify(_resource).examineStorageSubSystemCommandNfsVersion(storageCmd);     //Only 1 time, default for verify.
+    }
+
+    @Test
+    public void testStorageNfsVersionNotPresent(){
```

Comment:
I believe this method could be simpler. 
If it wants to test what is coded now. It would only need check if "_storageNfsVersion == null", then, Mockito.verify(mock, times(0)).examineStorageSubSystemCommandNfsVersion

Also for the other case you can use Mockito.doNothing, when calling "examineStorageSubSystemCommandNfsVersion". This is an integration test, you do not need to test the whole execution flow, you only need to test if the method (the one being tested) is calling the methods it should.

That means, to test the "checkStorageProcessorAndHandlerNfsVersionAttribute", you only need to worry about the "examineStorageSubSystemCommandNfsVersion" method.